logp/defaults_test.go

```diff
@@ -0,0 +1,111 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logp_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// TestDefaultConfig tests the default config ensuring the default
+// behaviour is to log to files
+func TestDefaultConfig(t *testing.T) {
+	cfg := logp.DefaultConfig(logp.DefaultEnvironment)
+
+	// Set cfg.Beat to avoid a log file like '-202405029.ndjson'
+	cfg.Beat = t.Name()
+
+	// Set Files.Path to an empty folder so we can assert the
+	// creation of the log file without being affected by other
+	// test runs.
+	cfg.Files.Path = t.TempDir()
+
+	if err := logp.Configure(cfg); err != nil {
+		t.Fatalf("did not expect an error calling logp.Configure: %s", err)
+	}
+
+	// Get the logger and log anything
+	logger := logp.L()
+	defer logger.Close()
+
+	logger.Info("foo")
+	_, fileName, lineNum, _ := runtime.Caller(0)
+	lineNum-- // We want the line number from the log
+	fileName = filepath.Base(fileName)
```

Comment:
Why do we need this here? This is really a test of what the `logger.Info` calls does isn't it? 

The point of this test is to prove that the default configuration logs to files, not to also test exact format of the logs themselves? Is that correct?

I think I am mostly concerned that this test succeeding currently depends on the exact placement of these lines in the file. It seems unnecessary to maintain this to verify what you need to.

Reply:
> The point of this test is to prove that the default configuration logs to files, not to also test exact format of the logs themselves? Is that correct?

Yes, but as I wrote the test I thought it would be nice to ensure some other fields we expect are there as well. The file and line number are essential exactly which bit of the code is producing the log entry. They're the first thing I look at when I'm dealing with a log entry I'm not used to see.

I don't mind removing this assertion, however I believe we should have a test that ensures our expected default behaviour, which includes some essential fields.

> I think I am mostly concerned that this test succeeding currently depends on the exact placement of these lines in the file. It seems unnecessary to maintain this to verify what you need to.

I get that, hence I'm reading  them from the runtime instead of just hard-coding them. As long as L52-54 are always together, no matter where in the file the tests will succeed.

I prefer adding a comment clarifying what those lines are doing and stating they must stay together rather than removing this assertion.

However, if you insist, I'm not against removing them for now and later writing a more comprehensive test with our expectations for a default log entry.

Reply:
I don't hate it, I just wanted to sanity check how necessary this was. The root of my concern is that this could be fragile, and also if this functionality is already tested as a core part of the upstream zap logger it uses underneath. 

It's cheap enough to have here, it is not worth going on a separate adventure to perfectly test every aspect of the logger right now.